src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -335,34 +337,47 @@ def postprocess(self, model_outputs, decoder_kwargs: Optional[Dict] = None, retu
         if self.type == "ctc_with_lm":
             if decoder_kwargs is None:
                 decoder_kwargs = {}
-            text = self.decoder.decode_beams(items, **decoder_kwargs)[0][0]
+            beams = self.decoder.decode_beams(items, **decoder_kwargs)
+            text = beams[0][0]
+            if return_timestamps:
+                # Simply cast from pyctcdecode format to wav2vec2 format to leverage
+                # pre-existing code later
+                chunk_offset = beams[0][2]
+                word_offsets = []
+                for word, (start_offset, end_offset) in chunk_offset:
+                    word_offsets.append({"word": word, "start_offset": start_offset, "end_offset": end_offset})
 
         else:
             skip_special_tokens = self.type != "ctc"
             text = self.tokenizer.decode(items, skip_special_tokens=skip_special_tokens)
             if return_timestamps:
-                if return_timestamps == "char":
-                    decoded = self.tokenizer.decode(
-                        items, skip_special_tokens=skip_special_tokens, output_char_offsets=True
-                    )
-                elif return_timestamps == "word":
-                    decoded = self.tokenizer.decode(
-                        items, skip_special_tokens=skip_special_tokens, output_word_offsets=True
-                    )
-                chunks = []
-                for item in decoded[f"{return_timestamps}_offsets"]:
-                    start = (
-                        item["start_offset"]
-                        * self.model.config.inputs_to_logits_ratio
-                        / self.feature_extractor.sampling_rate
-                    )
-                    stop = (
-                        item["end_offset"]
-                        * self.model.config.inputs_to_logits_ratio
-                        / self.feature_extractor.sampling_rate
+                char_offsets = self.tokenizer.decode(
+                    items, skip_special_tokens=skip_special_tokens, output_char_offsets=True
+                )["char_offsets"]
+                if return_timestamps == "word":
+                    word_offsets = self.tokenizer._get_word_offsets(
+                        char_offsets, self.tokenizer.replace_word_delimiter_char
                     )
-                    chunks.append({"text": item[return_timestamps], "timestamp": (start, stop)})
-                optional["chunks"] = chunks
+
+        if return_timestamps:
+            if return_timestamps == "word":
+                offsets = word_offsets
+            else:
+                offsets = char_offsets
+            chunks = []
+            for item in offsets:
+                start = (
+                    item["start_offset"]
+                    * self.model.config.inputs_to_logits_ratio
+                    / self.feature_extractor.sampling_rate
+                )
```

Comment:
```suggestion
                start = item["start_offset"] * self.model.config.inputs_to_logits_ratio
                start = start  / self.feature_extractor.sampling_rate
```
Since it's going to be split anyway, why not use several lines?

Reply:
I am fine with that suggestion, making the change.

Reply:
Went for `start /= self.feature_extractor.sampling_rate`  even if you don't have objection.